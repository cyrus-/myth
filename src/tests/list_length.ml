let s = "
type nat =
  | O
  | S of nat

type list =
  | Nil
  | Cons of nat * list

let list_length : list -> nat |>
  { [] => 0
  | [0] => 1
  | [0;0] => 2 } = ?"
