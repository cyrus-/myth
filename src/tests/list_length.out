(* Generated Thu Nov 13 00:35:27 2014 *)
let list_length : list -> nat =
  let rec f1 (l1:list) : nat =
    match l1 with
      | Nil -> O
      | Cons (n1, l2) -> S (f1 l2)
  in
    f1
;;
